src/main/java/seedu/commands/BudgetCommand.java

```diff
@@ -120,13 +127,28 @@ private void addBudget(SpendingList spendings, double budget, String timeRange)
         }
     }
 
+    private static void checkUserBudgetEntered(double smallerBudget, double biggerBudget, String smallerBudgetType,
+            String biggerBudgetType) {
+        if (biggerBudget <= 0) {
+            throw new WiagiInvalidInputException(INVALID_AMOUNT + ENTER_BUDGET_MESSAGE);
+        } else if (biggerBudget > MAX_LIST_TOTAL_AMOUNT) {
+            throw new WiagiInvalidInputException("Amount must be lesser than " + MAX_LIST_TOTAL_AMOUNT
+                    + ENTER_BUDGET_MESSAGE);
+        }
+        if (biggerBudget < smallerBudget) {
+            throw new WiagiInvalidInputException(smallerBudgetType + " should not be larger than " + biggerBudgetType
+                    + "! " + ENTER_BUDGET_MESSAGE);
+        }
+    }
+
     private static void initialiseDailyBudget(SpendingList spendings) {
         double amount = UNINITIALISED_BUDGET;
         Ui.initialiseDailyBudgetMessage();
         while (amount == UNINITIALISED_BUDGET) {
             try {
                 String userDailyBudget = Ui.readCommand();
-                amount = CommandUtils.formatAmount(userDailyBudget, BUDGET_INITIALISE_FORMAT);
+                amount = CommandUtils.roundAmount(userDailyBudget, BUDGET_COMMAND_FORMAT);
+                checkUserBudgetEntered(0, amount, "", "daily Budget");
```

Comment:
can change the 0 used here for code readability

Reply:
resolved